exporters/zipkin/src/main/java/io/opentelemetry/exporters/zipkin/ZipkinSpanExporter.java

```diff
@@ -0,0 +1,206 @@
+            .timestamp(toEpochMicros(spanData.getStartEpochNanos()))
+            .duration(endTimestamp - startTimestamp)
+            .localEndpoint(localEndpoint);
+
+    if (spanData.getParentSpanId().isValid()) {
+      spanBuilder.parentId(spanData.getParentSpanId().toLowerBase16());
+    }
+
+    for (Map.Entry<String, AttributeValue> label : spanData.getAttributes().entrySet()) {
+      spanBuilder.putTag(label.getKey(), attributeValueToString(label.getValue()));
+    }
+    Status status = spanData.getStatus();
+    if (status != null) {
+      spanBuilder.putTag(STATUS_CODE, status.getCanonicalCode().toString());
+      if (status.getDescription() != null) {
+        spanBuilder.putTag(STATUS_DESCRIPTION, status.getDescription());
+      }
+      if (!status.isOk()) {
+        spanBuilder.putTag(STATUS_ERROR, status.getCanonicalCode().toString());
+      }
+    }
+
+    for (SpanData.TimedEvent annotation : spanData.getTimedEvents()) {
+      spanBuilder.addAnnotation(toEpochMicros(annotation.getEpochNanos()), annotation.getName());
+    }
+
+    return spanBuilder.build();
+  }
+
+  @Nullable
+  private static Span.Kind toSpanKind(SpanData spanData) {
+    // This is a hack because the Span API did not have SpanKind.
+    if (spanData.getKind() == Kind.SERVER
+        || (spanData.getKind() == null && Boolean.TRUE.equals(spanData.getHasRemoteParent()))) {
+      return Span.Kind.SERVER;
+    }
+
+    // This is a hack because the Span API did not have SpanKind.
+    if (spanData.getKind() == Kind.CLIENT || spanData.getName().startsWith("Sent.")) {
+      return Span.Kind.CLIENT;
+    }
+
+    if (spanData.getKind() == Kind.PRODUCER) {
+      return Span.Kind.PRODUCER;
+    }
+    if (spanData.getKind() == Kind.CONSUMER) {
+      return Span.Kind.CONSUMER;
+    }
+
+    return null;
+  }
+
+  private static long toEpochMicros(long epochNanos) {
+    return MICROSECONDS.convert(epochNanos, NANOSECONDS);
+  }
+
+  private static String attributeValueToString(AttributeValue attributeValue) {
+    Type type = attributeValue.getType();
+    switch (type) {
+      case STRING:
+        return attributeValue.getStringValue();
+      case BOOLEAN:
+        return String.valueOf(attributeValue.getBooleanValue());
+      case LONG:
+        return String.valueOf(attributeValue.getLongValue());
+      case DOUBLE:
+        return String.valueOf(attributeValue.getDoubleValue());
+    }
+    return "";
+  }
+
+  @Override
+  public ResultCode export(final Collection<SpanData> spanDataList) {
+    List<byte[]> encodedSpans = new ArrayList<>(spanDataList.size());
+    for (SpanData spanData : spanDataList) {
+      encodedSpans.add(encoder.encode(generateSpan(spanData, localEndpoint)));
+    }
+    try {
+      sender.sendSpans(encodedSpans).execute();
+    } catch (IOException e) {
+      return ResultCode.FAILED_NOT_RETRYABLE;
+    }
+    return ResultCode.SUCCESS;
+  }
+
+  @Override
+  public void shutdown() {}
```

Comment:
URLConnection doesn't need to close on shutdown. However, the builder method that sets sender could warn the caller (in javadoc) that they are responsible to close the sender (a better choice than us closing it randomly)

Eventhough the calls here are synchronous, most senders will have resources open like kafka connections.

Reply:
you don't think that we should close the sender in the shutdown? Shutdown is only called when the exporter will no longer be used.

Reply:
I've always been of the opinion of "you open it, you close it" so, if someone hands you a resource (they opened) you don't close it, as that's their job. Ex if in spring a closeable bean is managed where it is defined and would be odd to have something else close it randomly.

if you do close it, then make sure that's known in the javadoc etc, so that it doesn't feel random and people can wrap if they don't want it closed.

Reply:
`Sender` doesn't appear to have a `shutdown` method, so I guess this is moot.  :)

Reply:
oh derp. it has a `close()`. nm me.

Reply:
It's a little tricky, since if you use the configuration option that creates a `Sender` in the background, then the user doesn't have control over the lifecycle. I'll change this to `close` the Sender and add javadoc to the configuration to let people know it'll happen.

Reply:
makes sense! you'd have to either do that or register a shutdown hook. I know we have one scenario somewhere where we do similar to what you said.